float32 linX
float32 linY
float32 linZ

float32 rotX
float32 rotY
float32 rotZ

float32 angX
float32 angY
float32 angZ